cmd/layotto/main.go

```diff
@@ -373,6 +385,24 @@ var cmdStart = cli.Command{
 	},
 }
 
+// ExtensionsRegister for register mosn rpc extensions
+func ExtensionsRegister(c *cli.Context) {
+	// tracer driver register
+	trace.RegisterDriver("SOFATracer", trace.NewDefaultDriverImpl())
+	// xprotocol action register
+	xprotocol.ResgisterXProtocolAction(xstream.NewConnPool, xstream.NewStreamFactory, func(codec api.XProtocolCodec) {
+		name := codec.ProtocolName()
+		trace.RegisterTracerBuilder("SOFATracer", name, xtrace.NewTracer)
+	})
+	// register protocols that are used by layotto.
+	_ = xprotocol.RegisterXProtocolCodec(&bolt.XCodec{})
+	_ = xprotocol.RegisterXProtocolCodec(&dubbo.XCodec{})
+	// register tracer
+	xtrace.RegisterDelegate(bolt.ProtocolName, tracebolt.Boltv1Delegate)
+	trace.RegisterTracerBuilder("SOFATracer", protocol.HTTP1, tracehttp.NewTracer)
```

Comment:
Could u please provide a mosn document to describe how users can register their extensions,and what do these abstractions mean? e.g. What is `RegisterDriver`,`ResgisterXProtocolAction ` and `RegisterXProtocolCodec ` ?

I don't understand this function  😢

Reply:
MOSN implement a group of trace drivers, but only a configured driver will be loaded.
A tracer driver can create different tracer by different protocol.
When MOSN receive a request stream, MOSN will try to start a tracer by the request protocol

Reply:
ResgisterXProtocolAction is a xprotocol framework's extensions. when a xprotocol implementation (defined by api.XProtocolCodec) registered, the registered action will be called. 

RegisterXProtocolCodec add a new xprotocol implementation, which is a wrapper for protocol register